fix: update firebase_auth example to not be dependent on an emulator

seanwestfall opened the pull request:
## Description

The main `flutterfire_auth example`, when compiled either on a device or through web will throw a connection error without this line commented out:

```
A network error (such as timeout, interrupted connection or unreachable host) has occurred.
```

removing the emulator setting, removes this error, helping people out there that are trying out this example.

## Related Issues

(https://github.com/firebase/firebase-ios-sdk/issues/7864)](https://github.com/firebase/firebase-ios-sdk/issues/7864)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

nilsreichardt commented on packages/firebase_auth/firebase_auth/example/lib/main.dart:
```diff
@@ -31,7 +31,8 @@ Future<void> main() async {
   }
 
-  await FirebaseAuth.instance.useAuthEmulator('localhost', 9099);
+  /* Uncomment the line below if you are running the example from a emulator, otherwise this line will cause a connection error */
```
Would be nice to avoid bloc comments: https://dart.dev/guides/language/effective-dart/documentation#dont-use-block-comments-for-documentation

russellwheatley replied:
Hey @seanwestfall, thanks for the PR! Please remove block comment and follow the same implementation as [firestore example](https://github.com/firebase/flutterfire/blob/master/packages/cloud_firestore/cloud_firestore/example/lib/main.dart#L10-L12) regarding setting/unsetting emulator 👍

seanwestfall commented:
@russellwheatley @nilsreichardt So I removed the block comments for the style guides comment style. 

I'm still running the tests for this PR, and that line for the emulator might be necessary for some of the tests to pass, so I might add more to the comments documentation, but the question might come down to whether allowing this to compile to a working example, or whether it should work for the tests, or not. Though some type of explanation should be included in either case.

russellwheatley commented:
@seanwestfall, that line is nothing to do with tests. We set up auth tests [here](https://github.com/firebase/flutterfire/blob/master/tests/test_driver/firebase_auth/firebase_auth_e2e.dart#L21) which requires auth emulator to be running. 

As mentioned previously, please use a variable to control auth emulator usage ([like Firestore example app](https://github.com/firebase/flutterfire/blob/master/packages/cloud_firestore/cloud_firestore/example/lib/main.dart#L10-L12)) and have it set to `false`. This will help folks understand what it is, and switch it in/out if need be. Thanks 🙏

seanwestfall commented:
@russellwheatley So I updated it, to use a boolean to check to see if the emulator is desired, with a default case set to false. So that when this is compiled, either for web or mobile, it will not cause a network error.

I changed the wording a little too, so it doesn't use any firestore specific wording, and the link goes to the firebase auth guidelines for emulator usage.
